src/rt/osx_tls.c

```diff
@@ -1,51 +0,0 @@
-/**
- * Helpers for determining TLS memory ranges on OS X.
- *
- * This unfortunately cannot be entirely done in D, as the OS X API uses
- * the Apple-specific blocks C extension.
- *
- * Copyright: David Nadlinger, 2012.
- * License:   <a href="http://www.boost.org/LICENSE_1_0.txt">Boost License 1.0</a>.
- * Authors:   David Nadlinger
- */
-
-#ifndef __APPLE__
-#ifndef __BLOCKS__
- #error "Need a C compiler with Apple Blocks support – not building on OS X with Clang?"
-#endif
-#endif
```

Comment:
The content of this file is moved into `sections_osx_x86_64.d`, which is already properly guarded for OS X 64bit.

The code which makes calling Blocks possible, `clang_block.d`, is cross-platform. Using that code only make sense if you have a C function taking Block you want to call. That C code needs to be compiled with Clang, and linking that on non-Apple platforms requires the Blocks runtime, which is noted in the documentation. We can't do anything, as far as I know, on the D side to enforce this.

If you try to compile the C code with something else than Clang you'll get a syntax error. Linking without the Blocks runtime will cause undefined symbol errors.